File: Daily Test/Python/Count Squares - Parallel To Axes 16.2.2026.py
n = int(input())

points = set()

for _ in range(n):
    x, y = map(int, input().split())
    points.add((x, y))

pts = list(points)

count = 0
size = len(pts)

for i in range(size):
    for j in range(i + 1, size):

        dx = pts[j][0] - pts[i][0]
        dy = pts[j][1] - pts[i][1]

        if abs(dx) == abs(dy) and dx != 0:

            p3 = (pts[i][0], pts[j][1])
            p4 = (pts[j][0], pts[i][1])

            if p3 in points and p4 in points:
                count += 1

print(count // 2)

File: Daily Test/Java/Count Squares - Parallel To Axes 16.2.2026.java
import java.util.*;

public class Main {

    static class Point {
        int x, y;
        Point(int x, int y) {
            this.x = x;
            this.y = y;
        }
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        HashSet<String> set = new HashSet<>();

        for (int i = 0; i < n; i++) {
            int x = sc.nextInt();
            int y = sc.nextInt();
            set.add(x + "," + y);
        }

        ArrayList<Point> pts = new ArrayList<>();

        for (String s : set) {
            String[] parts = s.split(",");
            pts.add(new Point(
                    Integer.parseInt(parts[0]),
                    Integer.parseInt(parts[1])
            ));
        }

        int count = 0;
        int size = pts.size();

        for (int i = 0; i < size; i++) {
            for (int j = i + 1; j < size; j++) {

                Point p1 = pts.get(i);
                Point p2 = pts.get(j);

                int dx = p2.x - p1.x;
                int dy = p2.y - p1.y;

                if (Math.abs(dx) == Math.abs(dy) && dx != 0) {

                    String p3 = p1.x + "," + p2.y;
                    String p4 = p2.x + "," + p1.y;

                    if (set.contains(p3) && set.contains(p4)) {
                        count++;
                    }
                }
            }
        }

        System.out.println(count / 2);
        sc.close();
    }
}